speechbrain/lobes/models/huggingface_transformers/huggingface.py

```diff
@@ -0,0 +1,404 @@
+
+        if is_local:
+            # Test for HuggingFace model
+            if any(File.endswith(".bin") for File in os.listdir(local_path)):
+                is_sb = False
+                return is_sb, checkpoint_filename, is_local
+
+            # Test for SpeechBrain model and get the filename.
+            for File in os.listdir(local_path):
+                if File.endswith(".ckpt"):
+                    checkpoint_filename = os.path.join(path, File)
+                    is_sb = True
+                    return is_sb, checkpoint_filename, is_local
+        else:
+            files = model_info(
+                path
+            ).siblings  # get the list of files of the Hub
+
+            # Test if it's an HuggingFace model or a SB one
+            for File in files:
+                if File.rfilename.endswith(".ckpt"):
+                    checkpoint_filename = File.rfilename
+                    is_sb = True
+                    return is_sb, checkpoint_filename, is_local
+
+            for File in files:
+                if File.rfilename.endswith(".bin"):
+                    checkpoint_filename = File.rfilename
+                    is_sb = False
+                    return is_sb, checkpoint_filename, is_local
+
+        err_msg = f"{path} does not contain a .bin or .ckpt checkpoint !"
+        raise FileNotFoundError(err_msg)
+
+    def _modify_state_dict(self, path, **args):
+        """A custom loading ensures SpeechBrain compatibility for pretrain and model
+        For example, wav2vec2 model pretrained with SB (Wav2Vec2Pretrain) has slightly different keys from Wav2Vec2.
+        This method handle the compatibility between the two.
+        Users should modify this function according to their own tasks.
+        Arguments
+        ---------
+        path : str
+            Checkpoint path, file name relative to the repo root.
+        """
+        return
+
+    def _load_sb_pretrained_parameters(self, path, modify_state_dict_fn):
+        """Loads the parameter of a HuggingFace model pretrained with SpeechBrain
+        and the HuggingFace Pretrain Object. It is necessary to perform a custom
+        loading because HuggingFace adds a level to the checkpoint when storing
+        the model breaking the compatibility Pretrain and model de/serialization.
+
+        For example, a typical Wav2Vec2 checkpoint for a given parameter
+        would be: model.conv.weight.data while for Wav2Vec2Pretrain it
+        is: model.wav2vec2.weight.data (wav2vec2 must be removed before loading).
+        Arguments
+        ---------
+        path : pathlib.Path
+            The full path to the checkpoint.
+        modify_state_dict_fn : method
+            A customized method which modifies the checkpoint's state_dict.
+        """
+        if modify_state_dict_fn is not None:
+            modified_state_dict = modify_state_dict_fn(path)
+        else:
+            modified_state_dict = torch.load(path, map_location="cpu")
```

Comment:
It seems that this line is never used. `modify_state_dict_fn` is always different to `None`